redaxo/src/addons/structure/lib/structure_context.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * @package redaxo5
+ */
+
+class rex_structure_context
+{
+    /**
+     * @var array
+     */
+    protected $params;
+
+    /**
+     * @param array $params
+     */
+    public function __construct(array $params = [])
+    {
+        if (isset($params['category_id'])) {
+            $params['category_id'] = rex_category::get($params['category_id']) instanceof rex_category ? $params['category_id'] : 0;
+
+            // Nur ein Mointpoint -> Sprung in die Kategorie
+            $mountpoints = $this->getMountpoints();
+            if (1 == count($mountpoints) && 0 == $params['category_id']) {
+                $params['category_id'] = current($mountpoints);
+            }
+        }
+
+        if (isset($params['article_id'])) {
+            $params['article_id'] = rex_article::get($params['article_id']) instanceof rex_article ? $params['article_id'] : 0;
+        }
+
+        if (isset($params['clang_id'])) {
+            $params['clang_id'] = rex_clang::exists($params['clang_id']) ? $params['clang_id'] : rex_clang::getStartId();
+
+            $stop = false;
+            if (rex_clang::count() > 1) {
+                if (!rex::getUser()->getComplexPerm('clang')->hasPerm($params['clang_id'])) {
+                    $stop = true;
+                    foreach (rex_clang::getAllIds() as $key) {
+                        if (rex::getUser()->getComplexPerm('clang')->hasPerm($key)) {
+                            $params['clang_id'] = $key;
+                            $stop = false;
+                            break;
+                        }
+                    }
+
+                    if ($stop) {
+                        echo rex_view::error('You have no permission to this area');
```

Comment:
Würde es sinn machen hier mit einer Exception zu arbeiten?

Reply:
Oder besser noch: das exit in der page lassen und nur die bedingung dafür hier im context checken